### PR TITLE
Test/viewmodel

### PR DESCRIPTION
**Pull request by 276746**
Add tests for ProfileViewModel and MapViewModel

image

### CODE REVIEW

File: app/src/test/java/com/github/wanderwise_inc/app/utils/MainDispatcherRule.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/utils/MainDispatcherRule.kt:1]: Lgtm, need to check use of Dispatchers but good way of separating it in a separate file if you need it for your tests

File: app/src/test/java/com/github/wanderwise_inc/app/viewmodel/MapViewModelTest.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/viewmodel/MapViewModelTest.kt:1]: Well done, nice way of testing and clean code. Can you maybe provide some comments on why using Dispatchers and other stuffs ?

File: app/src/test/java/com/github/wanderwise_inc/app/viewmodel/ProfileViewModelTest.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/viewmodel/ProfileViewModelTest.kt:1]: Lgtm, i think that some test that you wrote looks similar to the ProfileRepository tests but it's better to make more tests than not enough, well done. Clean code!

**Review by engjellismaili — COMMENTED** [review]
Overall really good code and good coverage achieved, well done ! As a remark i would say that somme comments on the way of testing could benefit other members to better understanding of tests.

File: app/src/test/java/com/github/wanderwise_inc/app/viewmodel/ProfileViewModelTest.kt
Comment (imaneb942) [inline app/src/test/java/com/github/wanderwise_inc/app/viewmodel/ProfileViewModelTest.kt:1]: Test Methods:
Well-structured and focused on specific functionalities.
Clear setup using @Before annotated method setup().
Test Cases:
Cover various scenarios of ProfileViewModel functionality.
Effectively utilize coroutines and Kotlin flows for asynchronous testing.
Mocking and Assertions:
Proper use of MockK's mocking and stubbing features.
Clear and descriptive assertions validate expected behavior.
Variable Naming:
Descriptive variable names follow good conventions.
Error Handling:
Error scenarios are appropriately handled, ensuring robustness.
Suggestions:
Consider adding comments for complex logic.
Ensure clear setup, execution, and verification steps for readability.

File: app/src/test/java/com/github/wanderwise_inc/app/viewmodel/MapViewModelTest.kt
Comment (imaneb942) [inline app/src/test/java/com/github/wanderwise_inc/app/viewmodel/MapViewModelTest.kt:1]: Test Structure:
Well-structured tests focusing on MapViewModel functionalities.
Proper setup using JUnit rules and annotations.
Coroutines and Flow:
Effective use of coroutines and flows for async testing.
Tests handle async behavior using runBlocking, flow, and test.
Mocking:
MockK is well-utilized for repository dependencies.
Stubbing with every and coEvery for controlled behavior.
Assertions:
Clear assertions validate expected behavior.
Method invocation verification is appropriate.
Dispatcher Management:
Dispatchers managed properly using Dispatchers.setMain and MainDispatcherRule.
Suggestions:
Consistent use of @OptIn(ExperimentalCoroutinesApi::class).
Consider adding comments for complex logic.

**Review by imaneb942 — APPROVED** [review]
overall great implementation, a bit of redundancy but otherwise it looks good!

**Pull request merged**